worldengine/hdf5_serialization.py

```diff
@@ -18,68 +18,68 @@ def save_world_to_hdf5(world, filename):
-    elevation_ths_grp["plain"] = world.layers['elevation'].thresholds[1][1]
-    elevation_ths_grp["hill"] = world.layers['elevation'].thresholds[2][1]
+    elevation_ths_grp["sea"] = world.elevation.thresholds[0][1]
+    elevation_ths_grp["plain"] = world.elevation.thresholds[1][1]
+    elevation_ths_grp["hill"] = world.elevation.thresholds[2][1]
     elevation_data = elevation_grp.create_dataset("data", (world.height, world.width), dtype=numpy.float)
-    elevation_data.write_direct(world.layers['elevation'].data)
+    elevation_data.write_direct(world.elevation.data)
 
     plates_data = f.create_dataset("plates", (world.height, world.width), dtype=numpy.uint16)
-    plates_data.write_direct(world.layers['plates'].data)
+    plates_data.write_direct(world.plates.data)
 
     ocean_data = f.create_dataset("ocean", (world.height, world.width), dtype=numpy.bool)
-    ocean_data.write_direct(world.layers['ocean'].data)
+    ocean_data.write_direct(world.ocean.data)
 
     sea_depth_data = f.create_dataset("sea_depth", (world.height, world.width), dtype=numpy.float)
-    sea_depth_data.write_direct(world.layers['sea_depth'].data)
+    sea_depth_data.write_direct(world.sea_depth.data)
 
     if world.has_biome():
         biome_data = f.create_dataset("biome", (world.height, world.width), dtype=numpy.uint16)
         for y in range(world.height):
             for x in range(world.width):
-                biome_data[y, x] = biome_name_to_index(world.layers['biome'].data[y][x])
+                biome_data[y, x] = biome_name_to_index(world.biome.data[y][x])
 
     if world.has_humidity():
         humidity_grp = f.create_group("humidity")
         humidity_quantiles_grp = humidity_grp.create_group("quantiles")
-        for k in world.layers['humidity'].quantiles.keys():
-            humidity_quantiles_grp[k] = world.layers['humidity'].quantiles[k]
+        for k in world.humidity.quantiles.keys():
+            humidity_quantiles_grp[k] = world.humidity.quantiles[k]
         humidity_data = humidity_grp.create_dataset("data", (world.height, world.width), dtype=numpy.float)
-        humidity_data.write_direct(world.layers['humidity'].data)
+        humidity_data.write_direct(world.humidity.data)
 
     if world.has_irrigation():
         irrigation_data = f.create_dataset("irrigation", (world.height, world.width), dtype=numpy.float)
-        irrigation_data.write_direct(world.layers['irrigation'].data)
+        irrigation_data.write_direct(world.irrigation.data)
 
     if world.has_permeability():
         permeability_grp = f.create_group("permeability")
         permeability_ths_grp = permeability_grp.create_group("thresholds")
-        permeability_ths_grp['low'] = world.layers['permeability'].thresholds[0][1]
-        permeability_ths_grp['med'] = world.layers['permeability'].thresholds[1][1]
+        permeability_ths_grp['low'] = world.permeability.thresholds[0][1]
+        permeability_ths_grp['med'] = world.permeability.thresholds[1][1]
```

Comment:
I would much prefer this when it comes to storing thresholds:
```python
for pair in world.permeability.thresholds:
    permeability_ths_grp[pair[0]] = pair[1]
```

As far as I understand the item at [x][0] is a naming string anyway. So far I haven't seen those strings used anywhere - so they could be renamed to 'low', 'med' etc., if necessary. Then the stored file would be the same as before, but the whole system could be expanded/changed more easily.